lib/retriever.js

```diff
@@ -161,16 +161,18 @@ function retrieve(options, callback){
       if(csvReg.test(record.file)){
         handleCsv(stream, record, scratchSpace, handleStream, handleStreamError);
       }else{
-        handleStream(retrieverPipeline(record, null, stream), record);
+        handleStream(record, null, stream);
       }
     }
   }
 
 
   function handleStreamError(record, err){
-    if(this.unpipe) this.unpipe();
-    if(this.destroy) this.destroy();
-    if(this.kill) this.kill();
+    if(this){
```

Comment:
When handleStreamError is provided as an errback to modules it might be called directly without a this-value being bound. This happens in `handleCsv.js` where a vrt file is internally created after the input stream has ended, but before the callback to `handleStream` is called.